### PR TITLE
Shooter Mk 2

### PR DESCRIPTION
**Pull request by AustinFortunato**
Shooter, better, faster, stronger- brought to you by mitsubishi

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/Shooter.java [hunk 1]
```diff
@@ -10,35 +10,36 @@
...
-    // inches/second
+    // inches/second/second
     public final static double GRAVITY = 386.2205;
     // inches
     public final static double TARGET_HEIGHT = 98.25;
+    private final double MAX_RPM = 5200;
     public final static double THETA = Math.toRadians(37);
-    public final static double DIAMETER = 4;
-    public final static double CIRCUMFERENCE = DIAMETER * Math.PI * Math.PI;
+    public final static double BALL_SPEED_RATIO = 27.358;
```
Comment (msoucy) [inline src/main/java/frc/robot/subsystems/Shooter.java:35]: Where'd this number come from?

File: src/main/java/frc/robot/subsystems/Shooter.java [hunk 2]
```diff
@@ -52,24 +53,28 @@ public CommandBase spinDown() {
...
+     * Finds the needed velocity to reach a target (x, y) or (horizontalDistance,
      * verticalDistance). The formula takes takes theta or launch angle, target and
      * gravity.
      */
-
     public static double calculateVelocity() {
-        if (distanceToTarget * Math.tan(THETA) >= verticalDistance) {
-            final double gravitySide = GRAVITY * distanceToTarget * distanceToTarget;
-            final double tanSide = distanceToTarget * Math.tan(THETA) - verticalDistance;
+        if (horizontalDistance * Math.tan(THETA) >= verticalDistance) {
```
Comment (msoucy) [inline src/main/java/frc/robot/subsystems/Shooter.java:89]: Where is `horizontalDistance` being set? It looks like it's being set once in the constructor, and not updating. Also `verticalDistance` and `distanceToTarget`, which is honestly the one that you probably need updated regularly.

**Review by msoucy — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/Shooter.java (hunk 1 above)
Comment (AustinFortunato) [reply src/main/java/frc/robot/subsystems/Shooter.java:35]: Basically gear ratios

File: src/main/java/frc/robot/subsystems/Shooter.java (hunk 2 above)
Comment (AustinFortunato) [reply src/main/java/frc/robot/subsystems/Shooter.java:89]: verticalDistance is a constant otherwise yes i will do those

**Review by msoucy — APPROVED** [review]
:ship: it

**Pull request merged**